### PR TITLE
Feat dc 18 time control

### PR DESCRIPTION
**Pull request by wikcioo**

### CODE REVIEW

File: CommunicationServer/Domain/DTOs/ResponseJoinedGameDto.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace Domain.DTOs;
+
+public class ResponseJoinedGameDto
```
Comment (M0Rek) [inline CommunicationServer/Domain/DTOs/ResponseJoinedGameDto.cs]: Consider a different name that says its a stream and not a response

File: CommunicationServer/Application/Entities/ChessTimer.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    private readonly PausableTimer _whiteTimer = new(1000.0);
+    private readonly PausableTimer _blackTimer = new(1000.0);
+    private readonly uint _timeControlSeconds;
+    private readonly uint _timeControlIncrement;
+    private readonly ValueWrapper<bool> _whitePlaying;
+    public double WhiteRemainingTimeMs { get; private set; }
+    public double BlackRemainingTimeMs { get; private set; }
+    
+    public delegate void EventHandler(object sender, EventArgs args, ResponseJoinedGameDto dto);
+    public event EventHandler ThrowEvent = delegate{};
```
Comment (M0Rek) [inline CommunicationServer/Application/Entities/ChessTimer.cs:18]: ThrowEvent is name suitable for throwing exceptions.

File: CommunicationServer/Application/Entities/ChessTimer.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        _whiteTimer.Start();
+        _blackTimer.Start();
+    }
+    
+    public void UpdateTimers()
+    {
+        if (_whitePlaying.Value)
+        {
+            _whitePlaying.Value = false;
+            WhiteRemainingTimeMs += _timeControlIncrement * 1000;
```
Comment (M0Rek) [inline CommunicationServer/Application/Entities/ChessTimer.cs:45]: Maybe store the increment in ms as you only use it that way

File: CommunicationServer/Application/Entities/GameRoom.cs [hunk 4]
```diff
@@ -136,11 +158,11 @@ MoveTypes.Castling or
...
 
-    public IObservable<MoveMadeDto> GetMovesAsObservable()
+    public IObservable<ResponseJoinedGameDto> GetMovesAsObservable()
     {
-        var newMoves = Observable.FromEvent<MoveMadeDto>(
-            x => MoveReceived += x,
-            x => MoveReceived -= x);
-        return _moves.ToObservable().Concat(newMoves);
+        var newMoves = Observable.FromEvent<ResponseJoinedGameDto>(
+            x => GameJoined += x,
```
Comment (M0Rek) [inline CommunicationServer/Application/Entities/GameRoom.cs:164]: That naming is confusing af. Is game joined happening many times?

File: CommunicationServer/Application/Application.csproj [hunk 5]
```diff
@@ -10,4 +10,10 @@
         <ProjectReference Include="..\Domain\Domain.csproj"/>
     </ItemGroup>
 
+    <ItemGroup>
+      <Reference Include="System.Reactive">
```
Comment (M0Rek) [inline CommunicationServer/Application/Application.csproj:14]: That packet referencing weirdly doesnt work on windows machines. Maybe import it the same way we import chessLib

**Review by M0Rek — COMMENTED** [review]
Hello good work make me kebab pls.

File: CommunicationServer/Domain/DTOs/ResponseJoinedGameDto.cs (hunk 1 above)
Comment (wikcioo) [reply CommunicationServer/Domain/DTOs/ResponseJoinedGameDto.cs]: Fixed, changed it to JoinedGameStreamDto

File: CommunicationServer/Application/Entities/ChessTimer.cs (hunk 2 above)
Comment (wikcioo) [reply CommunicationServer/Application/Entities/ChessTimer.cs:18]: I copied it from the internet where they used it for events. And the internet is always right

File: CommunicationServer/Application/Entities/ChessTimer.cs (hunk 3 above)
Comment (wikcioo) [reply CommunicationServer/Application/Entities/ChessTimer.cs:45]: _timeControlIncrement is a values received from the user when choosing the time control, which is in seconds. I suppose I could change it to ms

File: CommunicationServer/Application/Entities/GameRoom.cs (hunk 4 above)
Comment (wikcioo) [reply CommunicationServer/Application/Entities/GameRoom.cs:164]: Yup, fixed it by renaming to JoinedGameStream

File: CommunicationServer/Application/Application.csproj (hunk 5 above)
Comment (M0Rek) [reply CommunicationServer/Application/Application.csproj:14]: sure

File: CommunicationServer/Application/Entities/ChessTimer.cs (hunk 3 above)
Comment (M0Rek) [reply CommunicationServer/Application/Entities/ChessTimer.cs:45]: Ooor You could multiple it by a 1000 in a constructor

File: CommunicationServer/Application/Application.csproj (hunk 5 above)
Comment (wikcioo) [reply CommunicationServer/Application/Application.csproj:14]: Thanks, I appreciate it!

File: CommunicationServer/Application/Entities/ChessTimer.cs (hunk 3 above)
Comment (wikcioo) [reply CommunicationServer/Application/Entities/ChessTimer.cs:45]: Will do

**Review by M0Rek — APPROVED** [review]

**Pull request merged**